ironfish/src/workerPool/serializableMessage.ts

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import bufio from 'bufio'
+import { Serializable } from '../common/serializable'
+
+export enum WorkerMessageType {
+  BoxMessage = 'boxMessage',
+  CreateMinersFee = 'createMinersFee',
+  CreateTransaction = 'createTransaction',
+  GetUnspentNotes = 'getUnspentNotes',
+  MineHeader = 'mineHeader',
+  Sleep = 'sleep',
+  SubmitTelemetry = 'submitTelemetry',
+  TransactionFee = 'transactionFee',
+  UnboxMessage = 'unboxMessage',
+  VerifyTransaction = 'verifyTransaction',
+}
+
+export abstract class SerializableWorkerMessage implements Serializable {
+  constructor(id: number, type: WorkerMessageType) {
+    this.id = id
+    this.type = type
+  }
+
+  id: number
```

Comment:
nit: Can you move these fields above the constructor for consistency with the rest of the code base?

```ts
export abstract class SerializableWorkerMessage implements Serializable {
  id: number
  type: WorkerMessageType

  constructor(id: number, type: WorkerMessageType) {
    this.id = id
    this.type = type
  }

  ...
```

Reply:
Changed. 👍🏽